src/wms/src/main/java/org/geoserver/wms/CascadedLegendRequest.java

```diff
@@ -93,13 +106,13 @@ public JSONArray getCascadedJSONRules() {
             JSONArray cascadedRules =
                     layerLegends.getJSONObject(0).getJSONArray(JSONLegendGraphicBuilder.RULES);
             return cascadedRules;
-            // return jsonLegend;
         } catch (Exception e) {
+            LOGGER.log(Level.SEVERE, e.getMessage(), e);
+            throw new RuntimeException(e);
```

Comment:
Best to throw a service exception, with an indication of what you were doing there ("Error while cascading legend") and avoid logging, the Dispatcher will do that for you

Reply:
will do